packages/runtime/container-runtime/src/test/dataStoreContext.spec.ts

```diff
@@ -749,4 +751,147 @@ describe("Data Store Context Tests", () => {
+                0,
+                0);
+            summarizerNode.startSummary(0, new TelemetryNullLogger());
+
+            createSummarizerNodeFn = (
+                summarizeInternal: SummarizeInternalFn,
+                getGCDataFn: () => Promise<IGarbageCollectionData>,
+                getBaseGCDetailsFn: () => Promise<IGarbageCollectionDetailsBase>,
+            ) => summarizerNode.createChild(
+                summarizeInternal,
+                dataStoreId,
+                { type: CreateSummarizerNodeSource.Local },
+                // DDS will not create failure summaries
+                { throwOnFailure: true },
+                getGCDataFn,
+                getBaseGCDetailsFn,
+            );
+
+            const failingEntryPoint = new FluidObjectHandle<FluidObject>(
+                new LazyPromise(async () => {
+                    throw new Error("Simulating failure when initializing EntryPoint");
+                }),
+                "",
+                undefined as unknown as IFluidHandleContext,
+            );
+
+            factory = {
+                type: "store-type",
+                get IFluidDataStoreFactory() { return factory; },
+                instantiateDataStore: async (context: IFluidDataStoreContext, existing: boolean) =>
+                    provideDsRuntimeWithFailingEntrypoint
+                        ? new MockFluidDataStoreRuntime({ entryPoint: failingEntryPoint })
+                        : new MockFluidDataStoreRuntime(),
+            };
+            const registry: IFluidDataStoreRegistry = {
+                get IFluidDataStoreRegistry() { return registry; },
+                get: async (pkg) => (pkg === factory.type ? factory : undefined),
+            };
+            // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+            containerRuntime = {
+                IFluidDataStoreRegistry: registry,
+                on: (event, listener) => { },
+                logger: new TelemetryNullLogger(),
+            } as ContainerRuntime;
+        });
+
+        describe("Initialization", () => {
+            it("rejects ids with forward slashes", async () => {
+                const invalidId = "beforeSlash/afterSlash";
+                const codeBlock = () => new LocalDetachedFluidDataStoreContext({
+                    id: invalidId,
+                    pkg: [factory.type],
+                    runtime: containerRuntime,
+                    storage,
+                    scope,
+                    createSummarizerNodeFn,
+                    makeLocallyVisibleFn,
+                    snapshotTree: undefined,
+                    isRootDataStore: true,
+                });
+
+                assert.throws(codeBlock,
+                    (e: Error) => validateAssertionError(e, "Data store ID contains slash"));
+            });
+
+            describe("should error on attach if data object cannot be constructed", () => {
```

Comment:
@anthony-murphy a few new tests for the scenario you mentioned. I tried to exercise other code paths which should result in the same behavior but they are well-protected by asserts or other conditions that make it really awkward to construct tests to exercise them. If you can think of specific flows we want to add here, let me know.